File: 0045-jump-game-ii/0045-jump-game-ii.py

class Solution:
    def jump(self, nums: List[int]) -> int:
        ans, n = 0, len(nums)
        cp, mr = 0, 0
        
        for i in range(n - 1):
            mr = max(mr, i + nums[i])
            if i == cp:
                ans += 1
                cp = mr
        return ans

File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
         if(nums.size()<=1)return 0;
        int ans=0;
       int cp=0;
        int mr=0;
        for(int i=0;i<nums.size()-1;i++){
            mr=max(mr,nums[i]+i);
            if(i==cp){
                cp=mr;
                ans++;
            }
        }
        return ans;
    }
};